pkg/odo/cli/url/delete.go

```diff
@@ -41,7 +40,7 @@ func NewURLDeleteOptions() *URLDeleteOptions {
 // Complete completes URLDeleteOptions after they've been Deleted
 func (o *URLDeleteOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
 
-	if experimental.IsExperimentalModeEnabled() && util.CheckPathExists(o.DevfilePath) {
```

Comment:
I think we should still check for the existence of a devfile in the current context. Otherwise, this'll break `odo delete` for s2i components when experimental mode is enabled, fyi (currently, if experimental mode is enabled, we fall back to s2i support for commands when there's no devfile).